did_parser/src/did.rs

```diff
@@ -91,3 +93,9 @@ impl<'de> Deserialize<'de> for Did {
         Self::parse(did).map_err(serde::de::Error::custom)
     }
 }
+
+impl From<Did> for DidUrl {
+    fn from(did: Did) -> Self {
+        Self::from_did_parts(did.did().to_string(), 0..did.did.len(), did.method, did.id)
```

Comment:
I think the `did` field can be passed directly, even if the field is private, as this is in the same module with the struct.